src/github-repo.ts

```diff
@@ -229,6 +232,8 @@ export class GithubRepo {
     };
 
     await this.octokit.request(params);
+
+    return true;
```

Comment:
The only thing I’m a bit unsure about here is what this return value signifies. I’m guessing it’s success, but afaik failures are being communicated through exceptions anyway?

Reply:
This `waitFor` currently retries only if passed function returned `undefined`. That way the interface allows to return even when request succeeded but we didn't find expected value (like in case of get release by tag) and also allows us to fail fast if we got an error that is unexpected, like anything that is not a server error. The upload method doesn't fit this interface nicely, so I added this return here. We can change the flow to be controlled with errors though! Would this make more sense to you than the current implementation?

Reply:
> We can change the flow to be controlled with errors though! Would this make more sense to you than the current implementation?

Yes, but I also don’t want to add more work to a PR that is already perfectly fine as it is ;)

If not, I’d maybe make the return type `Promise<true>` rather than `Promise<boolean>`, I know it’s a small difference but it more clearly communicates that the function return value doesn’t have any meaning on its own and is instead just a constant

Reply:
Updated in https://github.com/mongodb-js/devtools-github-repo/pull/5/commits/21d93a55806639888ebad2ab7d272544cf09d251, does it look good to you?

Reply:
(I merged so I can release it and try it out, but happy to do some further refactoring here if you feel like it's needed!)